sklearn/ensemble/tests/test_partial_dependence.py

```diff
@@ -47,6 +46,24 @@ def test_partial_dependence_classifier():
     assert axes is None
     assert_array_equal(pdp, pdp_2)
 
+    # with trivial (no-op) sample weights
+    clf.fit(X, y, sample_weight=np.ones(len(y)))
+
+    pdp_w, axes_w = partial_dependence(clf, [0], X=X, grid_resolution=5)
+
+    assert pdp_w.shape == (1, 4)
+    assert axes_w[0].shape[0] == 4
+    assert_allclose(pdp_w, pdp)
+
+    # with non-trivial sample weights
+    clf.fit(X, y, sample_weight=sample_weight)
+
+    pdp_w2, axes_w2 = partial_dependence(clf, [0], X=X, grid_resolution=5)
+
+    assert pdp_w2.shape == (1, 4)
+    assert axes_w2[0].shape[0] == 4
+    assert np.min(np.abs(pdp_w2 - pdp_w)) / np.mean(np.abs(pdp_w)) > 0.25
```

Comment:
why not just the classic element-wise relative error check

```
np.abs(pdp_w2 - pdp_w) / np.abs(pdp_w) > something
```
?

Reply:
My concern is that the two partial dependences could happen to be very close at some points of the grid, so I want to check on the relative L^1 or L^2 distance between the two.
So I think `np.sum(np.abs(pdp_w2 - pdp_w)) / np.sum(np.abs(pdp_w)) > something` is actually a better alternative.

Reply:
but the test is deterministic right? At least it should be.

Reply:
Yes it is deterministic. The average / sum is taken accross deterministic points on the axis-grid defining the pdp.

Reply:
Are you referring to the fact that the threshold is too high on some distribution ? It is indeed a bit weird.

Reply:
>My concern is that the two partial dependences could happen to be very close at some points of the grid,

As the test is deterministic then if you try

```
np.abs(pdp_w2 - pdp_w) / np.abs(pdp_w) > something
```

and it passes then you're fine, as long as `something` isn't too small. The test cannot pass just "by chance".

If it fails (then you're out of luck), then considering the mean is an option.

Reply:
Indeed, the probability of passing the test by chance is lower with element-wise test.